### PR TITLE
WL-3961: 2 new db columns for reading list nested sections

### PR DESCRIPTION
**Pull request by nicholaswilson100**

### CODE REVIEW

**Comment by nicholaswilson100** [comment]
Constraints are another Jira

**Comment by nicholaswilson100** [comment]
I've removed the non null constraint from CITATION_ID because when we add, for example, a H1, then we will be inserting e.g.:
INSERT INTO CITATION_COLLECTION_ORDER (COLLECTION_ID,CITATION_ID,SECTION_TYPE, VALUE,LOCATION) 
VALUES (12345, null, 'heading1', 'Week 1', 1);

**Comment by nicholaswilson100** [comment]
As well as a migration script for us, do I need to commit a conversion sql for the community?

**Comment by buckett** [comment]
I'd be happy if these sort of issues are bundled up into larger requests as the schema may well end up changing if the UI is changed.

**Comment by buckett** [comment]
Yep, conversion script for the community should go in reference, there should be an 11 one in there already on the master branch.

File: citations-impl/impl/src/sql/oracle/sakai_citation.sql [hunk 1]
```diff
@@ -28,7 +28,9 @@ CREATE TABLE CITATION_CITATION
 CREATE TABLE CITATION_COLLECTION_ORDER
 (
 	COLLECTION_ID VARCHAR2 (36) NOT NULL,
-	CITATION_ID VARCHAR2 (36) NOT NULL,
+	CITATION_ID VARCHAR (36),
+	SECTION_TYPE ENUM('heading1','heading2', 'heading3', 'description'),
+	VALUE TEXT,
 	LOCATION NUMBER(10,0) NOT NULL
```
Comment (buckett) [inline citations-impl/impl/src/sql/oracle/sakai_citation.sql:34]: Is it worth adding a unique constraint on collection_id and location columns, so you can't end up with 2 items in the same position in the list?

**Comment by buckett** [comment]
Were you going to add the migration scripts to this pull request?

File: citations-impl/impl/src/sql/oracle/sakai_citation.sql (hunk 1 above)
Comment (nicholaswilson100) [inline citations-impl/impl/src/sql/oracle/sakai_citation.sql:34]: Yup I think so. Will do.

**Comment by nicholaswilson100** [comment]
Yes I'll add the migration scripts here too

**Comment by buckett** [comment]
Still no migration script here.

**Comment by nicholaswilson100** [comment]
Yep sorry been busying away with the front end - I'll do it today - and 
the conversion script

**Comment by nicholaswilson100** [comment]
I've attached the migration script to the jira

**Comment by nicholaswilson100** [comment]
For the conversion script, should I creating a pull request for my update to this file:
https://github.com/sakaiproject/sakai/blob/master/reference/docs/conversion/sakai_11_mysql_conversion.sql

**Comment by nicholaswilson100** [comment]
Migration script (not sure you can attach a file (possibly images) to a pull request):

ALTER TABLE CITATION_COLLECTION_ORDER ADD SECTION_TYPE ENUM('heading1','heading2', 'heading3', 'description', 'citation');
ALTER TABLE CITATION_COLLECTION_ORDER ADD VALUE TEXT;
ALTER TABLE CITATION_COLLECTION_ORDER MODIFY COLUMN CITATION_ID VARCHAR(36) null;

**Comment by buckett** [comment]
On 24 June 2015 at 14:37, Nick Wilson notifications@github.com wrote:

> For the conversion script, should I creating a pull request for my update
> to this file:
> 
> https://github.com/sakaiproject/sakai/blob/master/reference/docs/conversion/sakai_11_mysql_conversion.sql

I wouldn't create any pull requests for Sakai until there is a feature you
are wanting to push back.

## 

  Matthew Buckett, VLE Developer, IT Services, University of Oxford

**Comment by nicholaswilson100** [comment]
ok

**Comment by nicholaswilson100** [comment]
current URL: 
http://solo.bodleian.ox.ac.uk/PrimoWebServices/xservice/getit?onCampus=false&loggedIn=false&docId=oxfaleph015158494&institution=OXVU1

**Pull request merged**